src/pages/ContentScripts/features/repo-collection/Mydropdown.tsx

```diff
@@ -0,0 +1,309 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { Dropdown, Space, Button, Input, theme, Checkbox } from 'antd';
+import { DownOutlined } from '@ant-design/icons';
+import optionsStorage, {
+  HypercrxOptions,
+  defaults,
+} from '../../../../options-storage';
+import { getRepoName } from '../../../../helpers/get-repo-info';
+import Collection from './Collection';
+import { CheckboxChangeEvent } from 'antd/es/checkbox';
+
+const { useToken } = theme;
+
+const Mydropdown = () => {
+  const { token } = useToken();
+  const contentStyle: React.CSSProperties = {
+    backgroundColor: token.colorBgElevated,
+    borderRadius: token.borderRadiusLG,
+    boxShadow: token.boxShadowSecondary,
+  };
+  const menuStyle: React.CSSProperties = {
+    boxShadow: 'none',
+  };
+
+  const [open, setOpen] = useState(false);
+
+  const listFooter = (
+    <div className="text-right">
+      <Space>
+        <button className="btn" onClick={() => setOpen(false)}>
+          {' '}
+          返回
+        </button>{' '}
+        <button className="btn-primary btn" onClick={editRepo}>
+          {' '}
+          编辑
+        </button>{' '}
+      </Space>
+    </div>
+  );
+  const defaultCollection = {
+    Xlab2017: [
+      'X-lab2017/open-digger',
+      'X-lab2017/open-leaderboard',
+      'X-lab2017/open-wonderland',
+    ],
+    Hypertrons: ['hypertrons/hypertrons-crx', 'X-lab2017/open-leaderboard'],
+  };
+
+  const repoName = getRepoName();
+  const [options, setOptions] = useState<HypercrxOptions>(defaults);
+
+  const [footerContent, setFooterContent] = useState(listFooter);
+  const [items, setItems] = useState([
+    {
+      key: '1',
+      label: <div>Xlab2017</div>,
+    },
+    {
+      key: '2',
+      label: <div>Hypertrons</div>,
+    },
+  ]);
+  const [collectionData, setCollectionData] = useState(defaultCollection);
+  const [newCollectionName, setNewCollectionName] = useState('');
+  const [refresh, setRefresh] = useState(0);
+  const [isModalVisible, setModalVisible] = useState(false);
+  const [modalValue, setModalValue] = useState(['', ['']]);
+
+  const editFooter = (
+    <div className="text-right">
+      <Space style={{ padding: 8 }}>
+        <button className="btn" onClick={backToList}>
+          {' '}
+          返回
+        </button>{' '}
+        <Input
+          style={{ width: '180px' }}
+          placeholder="输入添加合集名称"
+          size="middle"
+          value={newCollectionName}
+          onChange={(e) => setNewCollectionName(e.target.value)}
```

Comment:
The first question here is that the `input` here is passed into the final `return` as one of the contents of the `editFooter`. But the `onChange `function here cannot be executed, which means that I cannot enter information in the `input`.

Reply:
I am not sure I fully understand your question. `onChange` event is supposed to be executed each time when you type something. I guess you want to know how you can execute functions defined in dropdown list in the `editFooter`. One way you can do is to pass a function as a parameter from the parent component (e.g. dropdown list here) to the child (e.g. editFooter) and add the function to the `onChange` events.

Here are example codes generated by GPT

Parent 
```
import React, { useState } from 'react';
import EditFooter from './EditFooter';

function DropdownList() {
  const [newCollectionName, setNewCollectionName] = useState('');

  const handleInputChange = (value) => {
    setNewCollectionName(value);
  };

  return (
    <div>
      {/* ... your dropdown list code ... */}
      <EditFooter onInputChange={handleInputChange} inputValue={newCollectionName} />
    </div>
  );
}

```

Child
```
import React from 'react';

function EditFooter({ onInputChange, inputValue }) {
  return (
    <div>
      <input
        placeholder="输入添加合集名称"
        size="middle"
        value={inputValue}
        onChange={(e) => onInputChange(e.target.value)}
      />
      {/* ... other footer content ... */}
    </div>
  );
}

export default EditFooter;

```

Please correct me if I am wrong. 😄

Reply:
I will have a try today! Thanks a lot 😃

Reply:
In React, `setState` triggers rerender, which means the functional component [will reexecutes](https://react.dev/learn/state-as-a-snapshot#rendering-takes-a-snapshot-in-time) from the beggining. So each time the component rerenders, the normal variable `editFooter` is a different object.

However, states are special variables, because they remain same in each render if there is no `setState` to set them to a new value. This rule also work when the state is an object variable. You can think that each `setState` will give the state a new version.

Back to your question. Actually both `onChange` and `setNewCollectionName` work as expected. Each `setNewCollectionName` triggers a rerender and the `newCollectionName` state changes too. Why the Input doesn't show those chars you just typed in?

Because the `footerContent` state remains the old version, whose value is the old `editFooter`. In your code, only `editRepo()` can `setFooterContent`.

---

The above explains the cause. But, this function should not be implemented in this way, which is totally a misuse of React state. Instead, a possible way is to use an `isEdit` state and [conditional rendering](https://react.dev/learn/conditional-rendering) to render the footer conditionally. DON'T store the JSX element into a state. 😆 

Always be careful with state and only use them when absolutely necessary.